internal/server/routes.go

```diff
@@ -229,12 +224,10 @@ func wrapRoute[Req, Res any](a *API, route route[Req, Res]) func(*gin.Context) e
 		}
 
 		req := new(Req)
-		if err := bind(c, req); err != nil {
+		if err := readRequest(c, req); err != nil {
 			return err
 		}
 
-		trimWhitespace(req)
```

Comment:
This moved into `readRequest` since they are both related to reading the request into the struct.